peer/brontide.go

```diff
@@ -3658,17 +3672,32 @@ func (p *Brontide) handleCloseMsg(msg *closeMsg) {
 		}
 
 		oShutdown.WhenSome(func(msg lnwire.Shutdown) {
-			p.queueMsg(&msg, nil)
+			// When we have a Shutdown to send, we defer it til the
+			// next time we send a CommitSig to remain spec
+			// compliant.
+			link.OnCommitOnce(htlcswitch.Outgoing, func() {
```

Comment:
IIUC, if the coast is already clear, then we can just send asap?

Reply:
OnCommitOnce will send synchronously if no commit_sig is owed.